requirements.txt

```
@@ -9,7 +9,8 @@ catalogue>=2.0.4,<2.1.0
 confection>=0.0.1,<1.0.0
 ml_datasets>=0.2.0,<0.3.0
 # Third-party dependencies
-pydantic>=1.7.4,!=1.8,!=1.8.1,<1.10.0
+pydantic>=1.7.4,!=1.8,!=1.8.1,<1.10.0; python_version < "3.7"
+pydantic>=1.7.4,!=1.8,!=1.8.1,<1.11.0; python_version >= "3.7"
```

You don't need the split requirements, pip just won't install newer versions if they're marked as python 3.7+.

```suggestion
pydantic>=1.7.4,!=1.8,!=1.8.1,<1.11.0
```